src/ast/passes/semantic_analyser.cpp

```diff
@@ -554,7 +554,17 @@ void SemanticAnalyser::visit(Call &call)
 
   if (call.func == "hist") {
     check_assignment(call, true, false, false);
-    check_nargs(call, 1);
+    if (!check_varargs(call, 1, 2))
+      return;
+    if (call.vargs->size() == 1) {
+      call.vargs->push_back(new Integer(0, call.loc)); // default bits is 0
```

Comment:
I don't think we should be modifying the AST in semantic analysis. I see one other occurrence in semantic analysis where we inject a cast, but I think that's bad too. Ideally mutable passes should derive from `Mutator` visitor so it's explicit.

In this case, I think it'd be better to teach the other passes to use a default value of 0 where appropriate